rel/app/env.sh.eex

```diff
@@ -8,7 +8,6 @@ export RELEASE_DISTRIBUTION="none"
 export MIX_ARCHIVES="${RELEASE_ROOT}/vendor/archives"
 export MIX_REBAR3="${RELEASE_ROOT}/vendor/rebar3"
 export LIVEBOOK_SHUTDOWN_ENABLED=${LIVEBOOK_SHUTDOWN_ENABLED:-true}
-export LIVEBOOK_DESKTOP=true
```

Comment:
@wojtekmach I did a search and I don't think we use this anywhere, instead we rely on `MIX_TARGET` at compile time. Let me know if I missed something!

Reply:
@josevalim actually you added it specifically in 75426e0d1aa2a87dd2c9fcc9deaaca02dc4bd95110d62cdcfc55c2dd01619b4eR11, do you remember why?

Reply:
Probably to make it easier to detect somewhere down below... do we have any shared initialization paths between the different execution modes?

Reply:
Ah, we were used to check it in the past for better error messages: https://github.com/chrismccord/livebook/blob/52240a167738f16a38d527f0bb0c571426c8b677/lib/livebook/application.ex#L202

Reply:
So yeah, we can nuke this. :)

Reply:
Ah, so basically the same purpose as this PR, but since we already have `MIX_TARGET`, we can rely on that :D